- all room-related canvases made singletons

AliceOfSNU opened the pull request:
check commit messages.

AliceOfSNU committed:
- all room-related canvases made singletons
- the lobby scene now can be used as main page(nickname, avatar select,,etc scene)
- dynamic determination of scene to load
- display currentRoomCanvas in loaded scene

AliceOfSNU commented:
what happened in commit #e0d886dd1b7de083e6ff6aa552da8ec6ff11674c

1. 지금은 어떤 scene을 불러올지 코드가 portal에만 쓰여 있는게 아니라 각종 canvas script에 박혀 있었는데 그거를 portal에서만 스트링으로 설정해 주면 코드가 반영해서 해당 씬 로드하도록 linkedSceneName 을 추가했습니다.
2. 모든 canvas를 singleton화 해서 scene이 바뀌어도 유지 되도록 했고, _roomsCanvases에 대한 call을 RoomCanvases.Instance 에 대한 singleton call 로 치환했습니다 (모든 call을 다 치환한 건 아닌데 다 치환하는 게 좋아 보입니다.그러면 각 canvas에서 _roomCanvases를 들고 있을 필요도 없고 first initialize도 필요가없습니다)
3. 게임 scene에서 그 룸의 currentRoomCanvas를 불러오는 기능을 다시 살렸습니다.

AliceOfSNU commented:
KNOWN BUGS:
- Lobby 에서 플레이 버튼 눌렀을때 Photon Connect가 실행 안되는 경우가 있음.. -> 왜지??? 지금 해결 방법은 다시 플레이 하면 됨.
- 아바타 controller가 마우스 lock 잡아버리면 UI안먹힘 -> lock 잡는기능을 강제로 꺼버릴 예정

AliceOfSNU committed:
- edited character settings - disabled mouse lock
- edited TestConnect.cs -> solved no connection problem

AliceOfSNU committed:
LoadCharacter has been also singleton-ized.
Now it works locally on other scenes as well.

AliceOfSNU committed:
allow networked instantiation of character. (only character1 has multiplayer controller now. will apply to others if this works well)

AliceOfSNU committed:
Merge branch '#48-new-avatar' into world

brainini commented:
그 외에도 여러 bug가 있네요 한준님 수정 필요해보입니다.

1. Avatar selection page에서 아바타 겹쳐 보임
2. ClickerGamePortal로 갔다가 방 생성 후 LeaveRoom 하면 mainroom으로 튕긴 후 시야 고정되고 멈춤
3. whiteboard portal로 들어가면 nullreference오류 발생
4. yacht game 누르고 닉네임 입력 후 join눌러도 Scene 변화 안됨.

AliceOfSNU commented:
아 portal interface 변경후 clicker game 아닌 다른 씬들에 적용이 안되어서 그렇네요 빨리 적용해 보도록 하겠습니다

AliceOfSNU committed:
UI 1. Avatar customization tabs

AliceOfSNU committed:
Merge branch '#48-new-avatar' into world

AliceOfSNU committed:
- Added Avatar customization tab view.
- updated to new Avatar

AliceOfSNU committed:
- now whiteboard scene loads correctly.
- known problem : Avatar size must scale according to scene's scale.

AliceOfSNU committed:
[feature]
- Added tabs selection

AliceOfSNU committed:
[feature]
- avatar controller auto decelerate
- press R to enable mouse input for rotation
- ESC to disable mouse input for view

AliceOfSNU committed:
[bugfix]
- on Back to Lobby, if already in lobby - scene, do not load scene again.
- set portal height to match character spawn height.

AliceOfSNU committed:
many erraneous behaviour fixed

AliceOfSNU committed:
Merge branch 'dev' into world

brainini committed:
[bugfix] roomlist refresh by TestConnect

brainini committed:
Merge pull request #56 from XREAL-SNU/world_roombugfix

[bugfix] roomlist refresh by TestConnect
